code/game/gamemodes/devil/game_mode.dm

```diff
@@ -0,0 +1,93 @@
+/datum/game_mode
+	var/list/datum/mind/sintouched = list()
+	var/list/datum/mind/devils = list()
+	var/devil_ascended = 0 // Number of arch devils on station
+
+/datum/game_mode/proc/auto_declare_completion_sintouched()
+	var/text = ""
+	if(sintouched.len)
+		text += "<br><span class='big'><b>The sintouched were:</b></span>"
+		var/list/sintouchedUnique = uniqueList(sintouched)
+		for(var/S in sintouchedUnique)
+			var/datum/mind/sintouched_mind = S
+			text += printplayer(sintouched_mind)
+			text += printobjectives(sintouched_mind)
+			text += "<br>"
+		text += "<br>"
+	world << text
+
+/datum/game_mode/proc/auto_declare_completion_devils()
+	/var/text = ""
+	if(devils.len)
+		text += "<br><span class='big'><b>The devils were:</b></span>"
+		for(var/D in devils)
+			var/datum/mind/devil = D
+			text += printplayer(devil)
+			text += printdevilinfo(devil)
+			text += printobjectives(devil)
+			text += "<br>"
+		text += "<br>"
+	world << text
+
+
+/datum/game_mode/proc/finalize_devil(datum/mind/devil_mind)
+
+	var/trueName= randomDevilName()
+
+	devil_mind.devilinfo = devilInfo(trueName, 1)
+	devil_mind.store_memory("Your devilic true name is [devil_mind.devilinfo.truename]<br>[lawlorify[LAW][devil_mind.devilinfo.ban]]<br>You may not use violence to coerce someone into selling their soul.<br>You may not directly and knowingly physically harm a devil, other than yourself.<br>[lawlorify[LAW][devil_mind.devilinfo.bane]]<br>[lawlorify[LAW][devil_mind.devilinfo.obligation]]<br>[lawlorify[LAW][devil_mind.devilinfo.banish]]<br>")
+	devil_mind.devilinfo.owner = devil_mind
+	devil_mind.devilinfo.give_base_spells(1)
+	spawn(10)
+		devil_mind.devilinfo.update_hud()
+	if(issilicon(devil_mind.current))
+		add_law_sixsixsix(devil_mind.current)
+
+/datum/game_mode/proc/add_devil_objectives(datum/mind/devil_mind, quantity)
+	var/list/validtypes = list(/datum/objective/devil/soulquantity, /datum/objective/devil/soulquality, /datum/objective/devil/sintouch, /datum/objective/devil/buy_target)
+	for(var/i = 1 to quantity)
+		var/type = pick(validtypes)
+		var/datum/objective/devil/objective = new type(null)
+		objective.owner = devil_mind
+		devil_mind.objectives += objective
+		if(!istype(objective, /datum/objective/devil/buy_target))
+			validtypes -= type //prevent duplicate objectives, EXCEPT for buy_target.
+		else
+			objective.find_target()
+
+/datum/mind
+	var/datum/devilinfo/devilinfo //Information about the devil, if any.
+
+/datum/mind/proc/announceDevilLaws()
+	if(!devilinfo)
+		return
+	current << "<span class='warning'><b>You remember your link to the infernal.  You are [src.devilinfo.truename], an agent of hell, a devil.  And you were sent to the plane of creation for a reason.  A greater purpose.  Convince the crew to sin, and embroiden Hell's grasp.</b></span>"
```

Comment:
on our codebase when you're display text to someone, you use this format:

`to_chat(thing displayed to, message)

this would be, for example, 
`to_chat(current, "<span class='warning'><b>You remember your link to the infernal.  You are [src.devilinfo.truename], an agent of hell, a devil.  And you were sent to the plane of creation for a reason.  A greater purpose.  Convince the crew to sin, and embroiden Hell's grasp.</b></span>")